From 2664376ca14dad1aaeb77dca84d4c8fcf5cf3f18 Mon Sep 17 00:00:00 2001
From: Michael-CY Lee <michael-cy.lee@mediatek.com>
Date: Wed, 19 Jun 2024 11:31:09 +0800
Subject: [PATCH 097/236] mtk: hostapd: do not consider bss ignore list when
 parsing ML parnter link

By definition, adding BSS into the ignore list only makes wpa_supplicant
tries the listed BSS with a low priority, but does not prevent the
listed BSS from being used.

It migth not be suitable to check the ignore list when parsing ML
partner links. It made wpa_supplicant associate with AP MLD by only
partial links with the link inside the ignore list being excluded.

After apply this commit, a partner link in the BSS ignore list will
not be ignored

Signed-off-by: Michael-CY Lee <michael-cy.lee@mediatek.com>
---
 wpa_supplicant/bss.c | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/wpa_supplicant/bss.c b/wpa_supplicant/bss.c
index ce5835623..d02dbe8b8 100644
--- a/wpa_supplicant/bss.c
+++ b/wpa_supplicant/bss.c
@@ -1699,9 +1699,12 @@ wpa_bss_parse_ml_rnr_ap_info(struct wpa_supplicant *wpa_s,
 				    (bss_params & (RNR_BSS_PARAM_SAME_SSID |
 						   RNR_BSS_PARAM_CO_LOCATED)) ||
 				    wpa_scan_res_match(wpa_s, 0, neigh_bss,
-						       ssid, 1, 0)) &&
+						       ssid, 1, 0))
+#if 0 /* MLD partner link should not be excluded */
 				   !wpa_bssid_ignore_is_listed(
-					   wpa_s, neigh_bss->bssid)) {
+					   wpa_s, neigh_bss->bssid)
+#endif
+				    ) {
 				struct mld_link *l;
 
 				bss->valid_links |= BIT(link_id);
-- 
2.45.2

